src/components/list/List.jsx

```diff
@@ -1,5 +1,79 @@
+import { useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { changeList } from '../../store/list/listSlice';
+import useListQueries from './listHooks';
+import { fetchRecipe } from '../../store/recipe/recipeSlice';
+
 function List() {
-    return <section>List</section>;
+    const curCuisine = useSelector((state) => state.list?.cuisine);
+    const search = useSelector((state) => state.list.searchParam);
+    const offsetNum = useSelector((state) => state.list.offset);
+    const total = useSelector((state) => state.list.total);
+
+    const { data, error, isLoading, refetch } = useListQueries();
+
+    useEffect(() => {
+        if (!curCuisine && !search) {
+            refetch();
+        }
+    }, [refetch, curCuisine, search]);
+
+    const dispatch = useDispatch();
+    const openRecipe = (recipeId) => {
+        dispatch(fetchRecipe({ id: recipeId }));
+    };
+
+    const buttonClicked = () => {
+        dispatch(
+            search
+                ? changeList({
+                      cuisine: '',
+                      offset: offsetNum + 15,
+                  })
+                : changeList({
+                      cuisine: curCuisine,
+                      searchParam: '',
+                      offset: offsetNum + 15,
+                  })
```

Comment:
here better to create let with default object what you should pass to changeList
than write if (search)... and add this field and after call changeList

Reply:
also, not really getting the idea of how to achieve this